numpy/_core/strings.py

```diff
@@ -25,6 +38,186 @@
 ]
 
 
+MAX = np.iinfo(np.int64).max
+
+
+def find(a, sub, start=0, end=None):
+    """
+    For each element, return the lowest index in the string where
+    substring ``sub`` is found, such that ``sub`` is contained in the
+    range [``start``, ``end``).
+
+    Parameters
+    ----------
+    a : array_like, with `np.bytes_` or `np.str_` dtype
+
+    sub : array_like, with `np.bytes_` or `np.str_` dtype
+        The substring to search for.
+
+    start, end : array_like, with any integer dtype
+        The range to look in, interpreted as in slice notation.
+
+    Returns
+    -------
+    y : ndarray
+        Output array of ints
+
+    See Also
+    --------
+    str.find
+
+    Examples
+    --------
+    >>> a = np.array(["NumPy is a Python library"])
+    >>> np.strings.find(a, "Python", 0, None)
```

Comment:
Sorry to be nitpicky, but I'd remove `, 0, None` from the example, since those are defaults.